Domain/IP Filter Refactor (CRASM-3594)

hawkishpolicy opened the pull request:
<!-- GitHub renders PRs such that soft line breaks are treated as hard
line breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  This is the reason for the markdownline
disable directive just below.

For more details see:
https://github.com/github/markup/issues/1050#issuecomment-294654762 -->
<!-- markdownlint-disable MD013 -->
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##
- Replaces existing Domain and IP search with a MUI Autocomplete based solution.
   - Reusable component that adjusts based on "search_field" prop:
      - passing "ip" to component returns:
         - IP autocomplete with a list of options based on IP address.
         - Selected IPs are displayed as a Checkbox controlled list.
         - Options narrow based on user input.
      - passing "name" to component returns:
         - Domain autocomplete with a list of options based on Domain name.
         - Selected Domains are displayed as a Checkbox controlled list.
         - Options narrow based on user input.
- Created new /search/domains endpoint:
   - Allows users to search for domain or IP in elasticsearch.
- Created unit tests for new endpoint.
- Created unit tests for new frontend functionality and components.
- Sample data for lcoal development now includes actual names and not just ip addresses.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##
- Improves UX and brings Domain and IP filters in line with Region and Org dropdowns.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##
- tested locally
- added frontend unit tests
- added backend unit tests
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## 📷 Screenshots (if appropriate) ##
Screen Recording 2026-01-16 at 9.55.49 AM.mov

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [ ] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).
- [ ] Create a pre-release (necessary if and only if the pre-release version was bumped).

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release (necessary if and only if the version was bumped).

hawkishpolicy committed:
Add DomainAndIPFilter component to DrawerInterior
- implemented for IP filtering

hawkishpolicy committed:
Added domain/ip search endpoint:

hawkishpolicy committed:
Add region?org filters to domain/ip search API

hawkishpolicy committed:
Add subdomain populate to sample data

hawkishpolicy committed:
Typo fix in create_sample_domain

hawkishpolicy committed:
Add unit tests for Domain/IP search endpoint

hawkishpolicy committed:
Edit helper text

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Domain/IP-Filter-Refactor-CRASM-2704

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Domain/IP-Filter-Refactor-CRASM-2704

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Domain/IP-Filter-Refactor-CRASM-2704

hawkishpolicy committed:
Unit tests:

- Added mockIps and mockDomains to test-utils for consistent test data.
- Created DomainAndIpFilter.test.tsx to cover domain and IP filtering logic.
  - Tests cover the following:
    - Rendering of Domain and IP Autocompletes.
    - Displaying list of options when clicking on Autocomplete.
    - Narrowing down options based on user input.
- Updated RegionAndOrganizationFilters.test.tsx to fix async handling and improve element selection.

hawkishpolicy committed:
Refine unit tests in RegionAndOrganizationFilters

- For Global Admin:
  - Add separate tests for rendering of org options.
  - Add test for narrowing of org options based on user input.

hawkishpolicy committed:
Add unit tests for Regional Admin:

- Add tests to verify:
  - IP autocomplete renders.
    - options render when opened.
    - options narrow based on input.
  - Domain autocomplete renders.
    - options render when opened.
    - options narrow based on input.

hawkishpolicy committed:
Add unit test for Global View Admin:

- Ensure the following:
  - IP autocomplete renders correctly.
  - IP options render when autocomplete is clicked.
  - Options narrow down based on user input.

hawkishpolicy committed:
Add unit tests for Global View user:

- Ensure the following:
  - Domain autocomplete renders correctly.
  - Options render upon interaction.
  - Options narrow down based on user input.

hawkishpolicy committed:
Update coverage thresholds

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Domain/IP-Filter-Refactor-CRASM-2704

hawkishpolicy committed:
Adjust unit test for domain search endpoint

- Reason for Change:
  - Regional Admins should have access to all domains regardless of domain since they now have access to all regions.
  - The previous test incorrectly assumed that Regional Admins would be unauthorized to access certain domains.
- Changes Made:
  - Updated the unit test `test_domains_search_autofill_endpoint_regional_unauth` to `test_domains_search_autofill_endpoint_regional_auth`.
  - Modified the assertions to check for a successful response (status code 200) instead of an unauthorized response (status code 403).

hawkishpolicy committed:
Clean up code.

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Domain/IP-Filter-Refactor-CRASM-2704

hawkishpolicy committed:
Update __snapshots__

ameliav commented on frontend/src/components/FilterDrawer/DomainAndIPFilter.tsx:
```diff
@@ -0,0 +1,419 @@
+          return [];
+        }
+      } catch (error) {
+        // console.error('Error fetching domain and IP search results:', error);
```
I think you can remove this or make it a logger statement

hawkishpolicy replied:
great catch.

hawkishpolicy replied:
resolved via [6db12d0](https://github.com/cisagov/XFD/pull/1416/commits/6db12d0b2251dd8bdd52ee4ea0a54aa118bf71a9)

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Domain/IP-Filter-Refactor-CRASM-2704

hawkishpolicy committed:
Changed console statement to logger statements

ameliav reviewed (dismissed):
LGTM!

jsalinasnttdata commented on frontend/src/components/FilterDrawer/DrawerInterior.tsx:
```diff
@@ -261,11 +262,11 @@ export const DrawerInterior: React.FC<Props> = (props) => {
           </Stack>
         </AccordionSummary>
         <AccordionDetails>
-          <TaggedArrayInput
```
Is this component used anywhere else. I did a quick look an was only able to fine it referenced here. We can likely nuke it if not in use.

hawkishpolicy replied:
I was going to remove it once the code was accepted.

hawkishpolicy replied:
But I completely agree.

hawkishpolicy replied:
resolved via [2e40280](https://github.com/cisagov/XFD/pull/1416/commits/2e40280d4522fc11f1b337e9df88b1ad81993a44)

cduhn17 reviewed (dismissed):
LGTM

hawkishpolicy committed:
Delete TaggedInput.tsx

- deprecated component now that it is replaced by MUI Autocomplete based component.

hawkishpolicy committed:
Changed endpoint from hardcoded to constant

hawkishpolicy committed:
Clean up index by removing TaggedInput export

cduhn17 reviewed (approved):
LGTM - review after additional changes.
